app/ui/browser-modern/actions/page-effects.js

```diff
@@ -111,6 +112,61 @@ export function toggleCurrentPageDevtools() {
   };
 }
 
+export function setPageZoomLevel(pageId, zoomLevel, doc = document) {
+  return {
+    type: EffectTypes.SET_PAGE_ZOOM_LEVEL,
+    webview: getWebViewForPageId(doc, pageId),
+    pageId,
+    zoomLevel,
+  };
+}
+
+export function setCurrentPageZoomLevel(zoomLevel) {
+  return (dispatch, getState) => {
+    dispatch(setPageZoomLevel(PagesSelectors.getSelectedPageId(getState()), zoomLevel));
+  };
+}
+
+export function performPageZoomIn(pageId) {
+  return (dispatch, getState) => {
+    let zoomLevel = PagesSelectors.getPageZoomLevel(getState(), pageId);
+    zoomLevel += PageState.ZOOM_LEVEL_10_PERCENT;
+    dispatch(setPageZoomLevel(pageId, zoomLevel));
+  };
+}
+
+export function performCurrentPageZoomIn() {
```

Comment:
Would be less code with a setZoom that took a value or a delta `setZoom(val)`/`setZoom({ delta: -10 })`, but already written, so up to you for maintainability